src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/Perf.Strings.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Text.Json
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Strings
+    {
+        private const int DataSize = 100_000;
+
+        private ArrayBufferWriter<byte> _arrayBufferWriter;
+
+        private string[] _stringArrayValues;
+        private byte[][] _stringArrayValuesUtf8;
+
+        [Params(true, false)]
+        public bool Formatted;
+
+        [Params(true, false)]
+        public bool SkipValidation;
+
+        public enum Escape
```

Comment:
Not sure if it would be worth it to split the benchmarks into with and without special characters (and remove the one-escaped case).

Reply:
I believe each case here is relevant.
1) Purely ASCII text
2) ASCII text with few "escaped" characters sprinkled in, say someone has a payload with "<ascii text> "someone's name" <more ascii text>"
3) Non-ASCII text which is escaped (for instance the entire text is chinese characters).

Not having case 2 doesn't test the performance of oscillating between ascii/no-escape and non-ascii/escape path. An implementation could be on the "fast" path on ascii and then degrade to the "slow" path as soon as the first non-ascii is hit. We want to measure the value in being smarter and  reverting back to the "ascii" path which is why option 2 provides value.

Reply:
Fair enough